pages/AddProduct.vue

```
@@ -0,0 +1,88 @@
+                    id="description" 
+                    class="block w-full h-20 mt-1 rounded-lg border-2"/>
+                <p class="mb-4 text-right"> {{ form.description.length }}</p>
+
+                <label for="image">Obrázek</label>
+                <input
+                    ref="fileInput" 
+                    @change="handleFileChange"
+                    id="image" 
+                    class="block file:bg-white file:font-semibold file:rounded-md file:w-full file:h-20 file:border-none file:cursor-pointer w-full h-20 mb-4 rounded-md border-2" 
+                    type="file"
+                    accept=".jpeg, .jpg, .png, .svg" >
+                <label for="price">Cena</label>
+                <input v-model="form.price" id="price" class="block mt-1 mb-5 rounded-md border-2" type="number">
+                <div class="flex item-center justify-center">
+                    <button
+                        @click="submitForm"
+                        class="px-5 py-2 rounded-md bg-slate-800 text-white font-semibold">Odeslat</button>
+                </div>
+            </form>
+        </div>
+    </div>
+</template>
+
+<script setup>
+    const form = ref({
+        title:"",
+        description:"",
+        image:"",
+        price: Number
+    })
+    const fileInput = ref()
+
+    const handleFileChange = () =>{
+        // Uložení obrázku 
+        form.image.value = '// file path'
+    }
+
+    const  submitForm = async () =>{
+        try{
+            const validation = validateForm(form);
+
+            if (validation !== true) {
+                console.log(validation) 
+            return;
+            }
+           const response = await $fetch('url',{
+            method:'POST',
+            body:{
+                ...form
+            }
+        }) 
+            console.log(response)
+        }catch(error){
+            console.log(error)
+        }
+    }
+
+ const validateForm = (form) => {
```

This function will never return `true`. Can you see the issue?

---

Also, I think this function can be defined a bit more ergonomically. Returning `true`-ish values for both valid and invalid forms seem weird to me. I would probably throw an error if the validation failed and catch it in the `trycatch` block.

---

Lastly, what is the downside and upside of using functions as constants in comparison to using
`function validateForm(form) { ... }`